src/ymlmd/yml2md.py

```diff
@@ -0,0 +1,162 @@
+"""yml2md Markdown table generation tool.
+
+Generates a Markdown table from the combined, normalized, and sorted output of normalize_yml.
+
+EXIT STATUS
+    This utility exits with one of the following values:
+    0   No error occurred.
+    >0  An error occurred.
+
+Usage:
+  yml2md [--log-level=LEVEL] <yml_file>
+  yml2md (-h | --help)
+
+Options:
+  -h --help              Show this message.
+  --log-level=LEVEL      If specified, then the log level will be set to
+                         the specified value.  Valid values are "debug", "info",
+                         "warning", "error", and "critical". [default: info]
+"""
+
+# Standard Python Libraries
+import logging
+import sys
+from typing import Any
+
+# Third-Party Libraries
+import dateparser
+import docopt
+from schema import And, Schema, SchemaError, Use
+import yaml
+
+# cisagov Libraries
+from mdyml import DEFAULT_CVE_ID
+
+from . import __version__
+
+Software = list[dict[str, Any]]
+
+
+def load(filename: str) -> Software:
+    """Load the YML into a Python dictionary."""
+    ans = None
+    with open(filename, "r") as f:
+        ans = yaml.safe_load(f)
+
+    return ans
+
+
+def calculate_status(software: Software) -> Software:
+    """Calculate status field for each software entry."""
+    for s in software:
+        # We are only worrying about the main (default) CVE for now.
+        default_cve = s["cves"][DEFAULT_CVE_ID]
+        if len(default_cve["affected_versions"]) == 0:
+            if len(default_cve["fixed_versions"]) == 0:
+                if default_cve["investigated"]:
+                    s["status"] = "Not Affected"
+                else:
+                    s["status"] = "Unknown"
+            else:  # fixed_versions is not empty
+                s["status"] = "Fixed"
+        else:  # affected versions is not empty
+            if len(default_cve["fixed_versions"]) == 0:
+                s["status"] = "Affected"
+            else:
+                s["status"] = "Fixed"
+    return software
+
+
+def generate_markdown(software: Software) -> None:
+    """Generate Markdown manually."""
+    header_fields = [
+        "Vendor",
+        "Product",
+        "Affected Versions",
+        "Patched Versions",
+        "Status",
+        "Vendor Links",
+        "Notes",
+        "References",
+        "Reporter",
+        "Last Updated",
+    ]
+
+    # Print header and delimiter rows
+    for row_contents in [header_fields, ["-" * len(field) for field in header_fields]]:
+        print(f'| {" | ".join(row_contents)} |')
+
+    # Print table converting lists to comma separated strings
```

Comment:
This doesn't seem correct.

```suggestion
    # Print table converting lists to Markdown table rows
```